R/add_md.R

```diff
@@ -0,0 +1,53 @@
+#' Add markdown content to an XML object
+#'
+#' @param body an XML object generated via {tinkr}
+#' @param md a string of new markdown to insert
+#' @param where the location in the markdown document to insert the new markdown
+#'
+#' @return a copy of the XML object with the markdown inserted.
+add_md <- function(body, md, where = 0L) {
+  new <- md_to_xml(md)
+  add_node_children(body, new, where)
+}
+
+# Adds children to a specific location and then copies the xml of the full
+# document. 
+add_node_children <- function(body, nodes, where = 0L) {
+  for (child in rev(nodes)) {
+    xml2::xml_add_child(body, child, .where = where)
+  }
+  copy_xml(body)
+}
+
+add_node_siblings <- function(node, nodes, where = "after", remove = TRUE) {
+  for(sib in rev(nodes)) {
+    xml2::xml_add_sibling(node, sib, .where = where)
+  }
+  if (remove) xml2::xml_remove(node)
+}
```

Comment:
why doesn't this function return a copy?

Reply:
Actually, I'm realizing that the companion function, `add_node_children()` should *not* return a copy. 

I do not return a copy here because I use `add_node_siblings()` in a loop so that I can modify multiple nodes in a single document. If I return a copy at this level, then I would have to iterate on that copy. 

For reference, here are all the places in the code where these functions are used. 

```r
R/add_md.R|10| add_node_children(body, new, where)
R/asis-nodes.R|84| add_node_siblings(math[bespoke][[i]], new_nodes[[i]], remove = TRUE)
R/asis-nodes.R|123| add_node_siblings(last_line, nodes, remove = TRUE)
R/asis-nodes.R|195| add_node_siblings(ticks[[i]], new_nodes[[i]], remove = TRUE)

```

Reply:
I've updated the other function so these both act the same.